### PR TITLE
BOP-2 GetWalletV2 PM client integration

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,4 +48,6 @@ Those are all Environment variables needed by the application:
 | ADB2C_POLICY_NAME                        | The Active Directory B2C policy name                                              | string |
 | ADB2C_ADMIN_GROUP_NAME                   | The Active Directory B2C group that identifies admin users                        | string |
 | IN_MEMORY_CACHE_TTL                      | A time to live value for in memory cache that stores user's groups                | number |
-| WALLET_API_URL                           | The PM getWalletV2 API URL                                                        | string |
\ No newline at end of file
+| CSTAR_API_URL                            | The base url of the PM service                                                    | string |
+| CSTAR_API_BASE_PATH                      | Tha base path for the getWalletV2 URL                                             | string |
```
Comment (pasqualedevita) [inline README.md:52]: The base url of the CStar Apim service
Tha base path for the CStar Apim URL

File: openapi/pm-api.yaml [hunk 2]
```diff
@@ -30,8 +30,11 @@ paths:
           description: Generic Error
 
 securityDefinitions:
-  basicAuth:
-    type: basic
+  SubscriptionKey:
+    type: apiKey
+    name: Ocp-Apim-Subscription-Key
+    in: header
+    description: The API key obtained through the developer portal or the getService operation.
```
Comment (pasqualedevita) [inline openapi/pm-api.yaml:37]:
```suggestion
    description: The API key obtained through the developer portal.
```

**Review by pasqualedevita — APPROVED** [review]
for me it's ok

File: GetPMWallet/handler.ts [hunk 3]
```diff
@@ -44,37 +78,143 @@ type IHttpHandler = (
...
+  _: WalletV2ListResponse
+): Either<t.Errors, Wallet> => {
+  return Wallet.decode({
+    data: fromNullable(_.data)
+      .map(walletList =>
+        walletList.reduce((acc, wallet) => {
+          const info = wallet.info;
+          if (
+            wallet.walletType === WalletTypeEnum.Card &&
+            WalletCardInfoInput.is(info)
```
Comment (AleDore) [inline GetPMWallet/handler.ts:95]: Is this double check needed? If not we can manage each branch with a `switch`

File: GetPMWallet/handler.ts [hunk 4]
```diff
@@ -44,37 +78,143 @@ type IHttpHandler = (
...
-    };
-    return taskEither
-      .of(ResponseSuccessJson({ data: [mockWallet1, mockWallet2] }))
-      .fold<IResponseSuccessJson<Wallet> | ErrorTypes>(err => {
+  return async (context, { fiscalCode }) => {
+    return tryCatch(
+      () =>
+        getWalletV2({
+          "Fiscal-Code": fiscalCode,
+          SubscriptionKey: subscriptionKey
```
Comment (AleDore) [inline GetPMWallet/handler.ts:168]: Why don't we encapsulate `SubscriptionKey` into a `withDefaults` in client definition?

File: GetPMWallet/handler.ts [hunk 5]
```diff
@@ -44,37 +78,143 @@ type IHttpHandler = (
...
+        }),
+      toError
+    )
+      .foldTaskEither(
```
Comment (AleDore) [inline GetPMWallet/handler.ts:172]: Why `foldTaskEither` here? Can we just chain ?

File: GetPMWallet/handler.ts (hunk 3 above)
Comment (BurnedMarshal) [reply GetPMWallet/handler.ts:95]: Yes, this check is needed because the typings of info is `object`.

File: GetPMWallet/handler.ts (hunk 5 above)
Comment (BurnedMarshal) [reply GetPMWallet/handler.ts]: Fixed in 95a58e5

File: GetPMWallet/handler.ts (hunk 4 above)
Comment (BurnedMarshal) [reply GetPMWallet/handler.ts]: Fixed in 95a58e5, I've done some refractory on how the Client is created

File: openapi/pm-api.yaml (hunk 2 above)
Comment (BurnedMarshal) [reply openapi/pm-api.yaml]: Fixed in 95a58e5

File: README.md (hunk 1 above)
Comment (BurnedMarshal) [reply README.md]: Fixed in 95a58e5

**Review by AleDore — APPROVED** [review]
LGTM

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**